.github/workflows/build+test.yml

```diff
@@ -0,0 +1,23 @@
+name: Build and test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Build project and run unit tests
+        run: ./gradlew build
```

Comment:
Maybe we should also add Gradle Cash to allow caching dependencies and build outputs to improve workflow execution time.
https://github.com/actions/cache/blob/main/examples.md#java---gradle
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows